src/transformers/models/mask2former/modeling_mask2former.py

```diff
@@ -0,0 +1,2444 @@
+@add_start_docstrings(
+    "The bare Mask2Former Model outputting raw hidden-states without any specific head on top.",
+    MASK2FORMER_START_DOCSTRING,
+)
+class Mask2FormerModel(Mask2FormerPreTrainedModel):
+    main_input_name = "pixel_values"
+
+    def __init__(self, config: Mask2FormerConfig):
+        super().__init__(config)
+        self.pixel_level_module = Mask2FormerPixelLevelModule(config)
+        self.transformer_module = Mask2FormerTransformerModule(
+            in_features=config.decoder_config.feature_size, config=config
+        )
+        self.is_train = config.is_train
+
+        self.post_init()
+
+    @add_start_docstrings_to_model_forward(MASK2FORMER_INPUTS_DOCSTRING)
+    @replace_return_docstrings(output_type=Mask2FormerModelOutput, config_class=_CONFIG_FOR_DOC)
+    def forward(
+        self,
+        pixel_values: Tensor,
+        pixel_mask: Optional[Tensor] = None,
+        output_hidden_states: Optional[bool] = None,
+        output_attentions: Optional[bool] = None,
+        return_dict: Optional[bool] = None,
+    ) -> Mask2FormerModelOutput:
+        r"""
+        Returns:
+            `Mask2FormerModelOutput`
+
+        Examples:
+        ```python
+        >>> import torch
+        >>> from PIL import Image
+        >>> import requests
+        >>> from transformers import Mask2FormerImageProcessor, Mask2FormerModel
+
+        >>> # download texting image
+        >>> url = "http://images.cocodataset.org/val2017/000000039769.jpg"
+        >>> image = Image.open(requests.get(url, stream=True).raw)
+
+        >>> # Load image preprocessor and Mask2FormerModel trained on ADE20K instance segmentation dataset
+        >>> image_processor = Mask2FormerImageProcessor.from_pretrained("facebook/mask2former-instance-swin-small-ade")
+        >>> model = Mask2FormerForUniversalSegmentation.from_pretrained("facebook/mask2former-instance-swin-small-ade")
+        >>> inputs = image_processor(image, return_tensors="pt")
+
+        >>> with torch.no_grad():
+        ...     outputs = model(**inputs)
+
+        >>> mask_predictions = outputs.transformer_decoder_mask_predictions
+        >>> class_predictions = outputs.transformer_decoder_class_predictions
+        ```
+        """
+
+        if pixel_values is None:
+            raise ValueError("You have to specify pixel_values")
+
+        output_attentions = output_attentions if output_attentions is not None else self.config.output_attentions
+        output_hidden_states = (
+            output_hidden_states if output_hidden_states is not None else self.config.output_hidden_states
+        )
+        return_dict = return_dict if return_dict is not None else self.config.use_return_dict
+
+        batch_size, _, height, width = pixel_values.shape
+
+        if pixel_mask is None:
+            pixel_mask = torch.ones((batch_size, height, width), device=pixel_values.device)
+
+        pixel_level_module_output: Mask2FormerPixelLevelModuleOutput = self.pixel_level_module(
```

Comment:
```suggestion
        pixel_level_module_output = self.pixel_level_module(
```
This is a bit clearer to me